test/tests/functional/pbs_passing_environment_variable.py

```diff
@@ -89,7 +89,28 @@ def test_passing_shell_function(self):
         Define a shell function with new line characters and check that
         the function is passed correctly
         """
-        os.environ['foo'] = '() { if [ /bin/true ]; then\necho hello;\nfi\n}'
+        # Check if ShellShock fix for exporting shell function in bash exists
+        # on this system and what "BASH_FUNC_" format to use
+        foo_scr = """#!/bin/bash
+foo() { a=B; echo $a; }
+export -f foo
+env | grep foo
+unset -f foo
+exit 0
+"""
+        fn = self.du.create_temp_file(body=foo_scr)
+        self.du.chmod(path=fn, mode=0755)
+        foo_msg = 'Failed to run foo_scr'
+        ret = self.du.run_cmd(self.server.hostname, cmd=fn)
+        self.assertEqual(ret['rc'], 0, foo_msg)
+        msg = 'BASH_FUNC_'
+        self.n = 'foo'
+        for m in ret['out']:
+            if m.find(msg) != -1:
+                self.n = m.split('=')[0]
+                continue
```

Comment:
shouldn't we break instead of continue?

Reply:
@prakashcv13 , I have made the continue into a break. Thanks for the sign off.